tools/devtools/utils.mjs

```diff
@@ -0,0 +1,262 @@
+import {readFileSync, writeFileSync, mkdirSync} from 'fs';
```

Comment:
At a glance all the methods in your tool that make file operations are `async` already so I'd vote to use the variants from `fs/promises`. That usually makes things easier to refactor later on as well and saves from probably unexpected halts